### PR TITLE
style: paint-order serialization fixes.

### PR DESCRIPTION
**Pull request by emilio**
See each commit for details.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18627)

### CODE REVIEW

File: components/style/properties/longhand/inherited_svg.mako.rs [hunk 1]
```diff
@@ -165,74 +171,83 @@ ${helpers.predefined_type("marker-end", "UrlOrNone", "Either::Second(None_)",
...
-      SpecifiedValue(NORMAL)
+        SpecifiedValue::normal()
     }
 
     impl SpecifiedValue {
-        pub fn bits_at(&self, pos: u8) -> u8 {
-            (self.0 >> pos * SHIFT) & MASK
+        fn order_at(&self, pos: u8) -> PaintOrder {
+            // Safe because PaintOrder covers all possible patterns.
+            unsafe { ::std::mem::transmute((self.0 >> pos * SHIFT) & MASK) }
```
Comment (nox) [inline components/style/properties/longhand/inherited_svg.mako.rs]: Why not just store a `PaintOrder` in `SpecifiedValue`?

Comment (emilio) [reply components/style/properties/longhand/inherited_svg.mako.rs]: (There are three packed values, so it's not clear to me how could I do that)

**Comment by nox** [comment]
Nice.

@bors-servo r+

**Pull request merged**